dulwich/client.py

```diff
@@ -679,7 +679,29 @@ def send_pack(self, path, determine_wants, generate_pack_contents,
         :raises UpdateRefsError: if the server supports report-status
                                  and rejects ref updates
         """
-        raise NotImplementedError(self.send_pack)
+        from dulwich.repo import Repo
+        from dulwich.protocol import ZERO_SHA
+        
+        target = Repo(path)
+        old_refs = target.get_refs()
+        if not old_refs:
+            old_refs["capabilities^{}"] = ZERO_SHA
```

Comment:
If the capabilities^ ref is leaked into the refs returned to the user somewhere, that is a bug. It is protocol specific, and an abstraction leak. I'll file a bug about this, and will file a bug with hg-git too about its dependency on this behaviour.